src/pages/MenuPage.js

```diff
@@ -24,7 +64,24 @@ export class MenuPage extends React.Component {
   }
 
   componentWillMount() {
-    const { database } = this.props;
+    const { database, settings } = this.props;
+
+    this.sections = SECTIONS;
+    // check is there are any modules enabled in this store customData
+    const usableModules = this.getUsedModules(settings.get(THIS_STORE_CUSTOM_DATA));
+    const usesModules = usableModules.length > 0;
+    this.sections.modules.buttons = usableModules;
+
+    if (!usesModules) {
+      this.layout = ORIGINAL_LAYOUT;
+      this.imageStyle = localStyles.originalImage;
+      this.sectionStyle = localStyles.section;
+    } else {
+      this.layout = VACCINE_MODULE_LAYOUT;
+      this.imageStyle = localStyles.VM_Image;
+      // Image will go on the left side as layout changes when modules are enabled
+      this.sectionStyle = { ...localStyles.section, flexDirection: 'row' };
```

Comment:
Am liking having `localStyles` be a function. Pass it values which can determine what styles are retrieved.  Abstracts a lot of style logic out

Reply:
it does look neater: https://github.com/openmsupply/mobile/pull/868/commits/cdc1d9034ff584383b6407ad421f324e2a2bf9f2, is this what you meant ?

Reply:
not sure if i like this if style getter is called frequently (i.e. def not for a table cell)

Reply:
Sorry what I meant was

```
const {someStyle} = localStyles({someProps})

const localStyles = ({someValueWhichAltersStyles}) => StyleSheet.create({
    someStyle: {
        color: someValueWhichAltersStyles ? 'white' : 'black'
    }
})
```

But they way you've done it is nice.

I'd also argue that doing it the above way, a simple function returning a value and evaluating simple conditionals could be faster than creating and mutating complex objects

Reply:
@joshxg, sorry I don't quite understand, how would your method deal with localStyles.section ? 
i.e. use the full localStyle.section but use flexDirection as row ?

I do see the use case for it with VM_Image, where we just want hight and width changed.

By the way, i don't see how the original way (which i think was the quickest way) and the get style mutate any object, they just create new objects by destructing localStyle and replacing optional fields

Reply:
Sorry originally it was more of just how I like to use `localStyles` as an actual function that abstracts away a lot of the style logic.

That's true about objects, just I think we're talking about tiny processor performance increases at this point, in which case dealing with primitive data types are pretty much always going to be faster than creating complex objects? 

Anyway this is more just off topic sorry